### DISCUSSION TITLE
fr/modes/train/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9409] [Docs]
# fr/modes/train/

Guide étape par étape pour entraîner les modèles YOLOv8 avec Ultralytics YOLO  , y compris des exemples d'entraînement sur une seule GPU et sur plusieurs GPU.

https://docs.ultralytics.com/fr/modes/train/

**Comment by Hamzaboughfir123** [comment]
I have a pretrained model trained on images to detect lamp light and know i want to use this model to train new images but know with different classes of lamp ?

**Reply by glenn-jocher** [reply]
Hey there! 👋 It's great that you're exploring training your pretrained model on new classes of lamps. To adapt your model for different lamp classes, you'll want to create a new dataset with annotations for the new classes. Make sure your dataset is properly labeled with the new classes and follow the training steps as you did for your initial model.

Here's a quick example on how to start training with your new dataset:

```python
from ultralytics import YOLO

# Load your existing model
model = YOLO('path/to/your/pretrained/model.pt')

# Train the model on your new dataset
results = model.train(data='your_new_dataset.yaml', epochs=100, imgsz=640)
```

Make sure 'your_new_dataset.yaml' is properly set up with paths to your new data and that it includes the new lamp classes you're introducing.

If you're adding more classes, remember to update your model's classification layer to account for the total number of classes (old + new). Happy training! 🚀

**Comment by NarymaneNanou** [comment]
I tried to train my model on my MacOs M1 using this command but there is an issue. 
results = model.train(data="coco8.yaml", epochs=100, imgsz=640, device="mps")

Pytorch does not implement orch.mps.get_mem_info() that you call in your code, so it stops the training

**Reply by glenn-jocher** [reply]
@NarymaneNanou thank you for reporting this MPS compatibility issue. While we fully support Apple Silicon training via `device="mps"`, some PyTorch memory functions are still in development. As a temporary solution, please try CPU training with `device="cpu"` until PyTorch implements this functionality. For more details see the Apple Silicon MPS Training section in our documentation.
